File: src-all/Python/S1461-check-if-a-string-contains-all-binary-codes-of-size-k.py
"""
Given a binary string s and an integer k.

Return True if every binary code of length k is a substring of s. Otherwise, return False.

 
Example 1:
Input: s = &quot;00110110&quot;, k = 2
Output: true
Explanation: The binary codes of length 2 are &quot;00&quot;, &quot;01&quot;, &quot;10&quot; and &quot;11&quot;. They can be all found as substrings at indicies 0, 1, 3 and 2 respectively.

Example 2:
Input: s = &quot;00110&quot;, k = 2
Output: true

Example 3:
Input: s = &quot;0110&quot;, k = 1
Output: true
Explanation: The binary codes of length 1 are &quot;0&quot; and &quot;1&quot;, it is clear that both exist as a substring. 

Example 4:
Input: s = &quot;0110&quot;, k = 2
Output: false
Explanation: The binary code &quot;00&quot; is of length 2 and doesn&#39;t exist in the array.

Example 5:
Input: s = &quot;0000000001011100&quot;, k = 4
Output: false

 
Constraints:
	1 <= s.length <= 5 * 10^5
	s consists of 0&#39;s and 1&#39;s only.
	1 <= k <= 20


"""
class Solution:
    def hasAllCodes(self, s: str, k: int) -> bool:
        
        pass


if __name__ == '__main__':
    assert Solution().hasAllCodes(0) == 0

File: src-all/C++/S1461-check-if-a-string-contains-all-binary-codes-of-size-k.cpp
/*
Given a binary string s and an integer k, return true if every binary code of length k is a substring of s. Otherwise, return false.

 
Example 1:
Input: s = "00110110", k = 2
Output: true
Explanation: The binary codes of length 2 are "00", "01", "10" and "11". They can be all found as substrings at indices 0, 1, 3 and 2 respectively.

Example 2:
Input: s = "0110", k = 1
Output: true
Explanation: The binary codes of length 1 are "0" and "1", it is clear that both exist as a substring. 

Example 3:
Input: s = "0110", k = 2
Output: false
Explanation: The binary code "00" is of length 2 and does not exist in the array.

 
Constraints:
	1 <= s.length <= 5 * 105
	s[i] is either '0' or '1'.
	1 <= k <= 20


*/
#include <cassert>
#include <iostream>

class Solution {
public:
    bool hasAllCodes(string s, int k) {
        
    }
};

int main() {
  Solution sol;
  assert(sol.hasAllCodes(0) == 0);
  std::cout << "Pass test cases!" << std::endl;
}
